org-roam-node.el

```diff
@@ -628,7 +628,7 @@ Assumes that the cursor was put where the link is."
         (goto-char (org-element-property :begin link))
         (when (and (org-in-regexp org-link-any-re 1)
                    (string-equal type "roam")
-                   (setq node (org-roam-node-from-title-or-alias path)))
+                   (setq node (save-match-data (org-roam-node-from-title-or-alias path))))
```

Comment:
In this case, I'll move the save-match-data into `org-roam-node-from-title-or-alias`.

Reply:
I don't mind, but IIUC, the idiomatic way of doing it in emacs appears
to put save-match-data in the function that needs the match data to be
saved.

At least, this it what I understand by reading at the documentation of
save-match-data:

```so ‘save-match-data’ should normally be used to save *your* match data
rather than your caller’s match data.```